docs/faq/index.rst

```diff
@@ -1,17 +1,276 @@
+
+* You can check the Docker container logs:
+
+  .. prompt:: bash $
+
+     docker logs <name or UID of Flocker container>
+
+For more information, see :ref:`flocker-logging`.
+
+My volume says it’s in a different state than I expected, why is this?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are a number of reasons this could happen.
+Your volume could be in detached, pending, attached, or deleted state.
+
+If your volume is in a different state than expected, then check the backend to see what is happening with your volume, or you can check the Flocker logs for any indication of issues or errors.
+
+If you continue to have errors, please contact support@clusterhq.com, where we may be able to help.
+
+The node running my container has frozen or locked up, what is going to happen?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the node your container is running has frozen or locked up, then Flocker may not be able to instruct the volume to move.
+The container using the volume may also not be able to stop, because no process may be able to tell it to stop.
+
+If your infrastructure allows it, you can restart or terminate your node whereby Flocker will be able to operate on the volume.
+If your infrastructure does not allow this, you may need wait to see if the node returns to being healthy.
+
+Why does the ``uft-flocker-volumes`` command not work?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This prefix ``uft`` was removed in favor of ``flockerctl``.
+For more information, see :ref:`flockerctl`.
+
+Using Flocker
+-------------
+
+How do I integrate Flocker with Docker Swarm?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. _the Flocker Google group: https://groups.google.com/forum/#!forum/flocker-users
+Flocker Swarm is Docker native!
+This means you can use our Docker volume driver to easily integrate with Swarm.
+
+Here are some useful links for more information:
+
+* :ref:`Using Flocker with Docker, Swarm, Compose <docker-integration>`
+* :ref:`about-docker-integration`
+
+
+How do I integrate Flocker with Kubernetes?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here are some useful links for more information:
+
+* :ref:`Using Flocker with Kubernetes <kubernetes-integration>`
+* `Demo: High Availability with Kubernetes and Flocker <https://clusterhq.com/2015/12/22/ha-demo-kubernetes-flocker/>`_
+* `Tutorial: Deploying a Replicated Redis Cluster on Kubernetes with Flocker <https://clusterhq.com/2016/02/11/kubernetes-redis-cluster/>`_
+
+How do I integrate Flocker with Mesos?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can integrate Flocker with Mesos using either the Mesos-Flocker Isolator, or with Marathon and the Flocker plugin for Docker.
+
+Here are some useful links for more information:
+
+* :ref:`Using Flocker with Mesos <mesos-integration>`
+* `Mesos-Flocker Isolator <http://flocker.mesosframeworks.com/ >`_
```

Comment:
This is a labs project (not even that, it's a demo), maybe we should mention that?